pallets/registrar/src/lib.rs

```diff
@@ -267,12 +278,15 @@ pub mod pallet {
             Ok(())
         }
 
-        /// Deregister parachain.
+        /// Deregister container-chain.
         ///
-        /// If a parachain is registered but not marked as valid_for_collating, this will remove it
+        /// If a container-chain is registered but not marked as valid_for_collating, this will remove it
         /// from `PendingVerification` as well.
         #[pallet::call_index(1)]
-        #[pallet::weight(10_000 + T::DbWeight::get().reads_writes(2,2).ref_time())]
+        #[pallet::weight(T::WeightInfo::deregister(
+            T::MaxGenesisDataSize::get(),
+            T::MaxLengthParaIds::get()
```

Comment:
We could make this a weight hint but I think it would over-complicate things, also in terms of UI. But I am open to suggestions. We can also add it in future PR

Reply:
We can add a separate `deregister_with_hints` extrinsic in the future, if the weight becomes a problem.

Reply:
hints are such an unpleasant UX/DX, I'd be in favor of avoiding them if possible.

Maybe we could have both: one extrinsic that assumes worst-case and refunds and another that takes a weight hint. That way we can at least offer users to choose their flavor of shit

Reply:
I think this is a nice idea yes